app/models/access_request.rb

```diff
@@ -0,0 +1,51 @@
+# == Schema Information
+#
+# Table name: access_request
+#
+#  id               :integer          not null, primary key
+#  email_address    :text
+#  first_name       :text
+#  last_name        :text
+#  organisation     :text
+#  reason           :text
+#  request_date_utc :datetime         not null
+#  requester_id     :integer
+#  status           :integer          not null
+#  requester_email  :text
+#
+
+class AccessRequest < ApplicationRecord
+  belongs_to :user, foreign_key: :requester_id, primary_key: :id
+
+  enum status: {
+    requested: 0,
+    approved: 1,
+    completed: 2,
+    declined: 3,
+  }
+
+  def admin?(current_user)
+    # should probably be using regex here?
+    admin_users = User.where("email ILIKE (?)", "%education.gov.uk")
+    admin_users.include?(current_user)
+  end
+
+  def update_access(access_request, requesting_user)
+    requested_user = User.find_by(email: access_request.email_address)
+    if requested_user == nil
+      access_request.create_requested_user(access_request, requesting_user, requested_user)
+    else
+      requested_user.update(organisations: requesting_user.organisations)
+    end
+  end
+
+  def create_requested_user(access_request, requesting_user, _requested_user)
```

Comment:
This might be nice to keep in a service object.

Regarding the arguments, how come we have an unused argument?